tests/python/gaia-ui-tests/gaiatest/gaia_test.py

```diff
@@ -642,14 +642,26 @@ def press_release_volume_up_then_down_n_times(self, n_times):
             };""", script_args=[n_times])
 
     def turn_screen_off(self):
-        self.marionette.execute_script("window.wrappedJSObject.ScreenManager.turnScreenOff(true)")
+        apps = GaiaApps(self.marionette)
+        self.marionette.switch_to_frame()
+        ret = self.marionette.execute_script("window.wrappedJSObject.ScreenManager.turnScreenOff(true)")
+        apps.switch_to_displayed_app()
+        return ret
 
     def turn_screen_on(self):
-        self.marionette.execute_script("window.wrappedJSObject.ScreenManager.turnScreenOn(true)")
+        apps = GaiaApps(self.marionette)
+        self.marionette.switch_to_frame()
+        ret = self.marionette.execute_script("window.wrappedJSObject.ScreenManager.turnScreenOn(true)")
+        apps.switch_to_displayed_app()
+        return ret
 
     @property
     def is_screen_enabled(self):
-        return self.marionette.execute_script('return window.wrappedJSObject.ScreenManager.screenEnabled')
+        apps = GaiaApps(self.marionette)
+        self.marionette.switch_to_frame()
+        ret = self.marionette.execute_script('return window.wrappedJSObject.ScreenManager.screenEnabled')
+        apps.switch_to_displayed_app()
+        return ret
 
```

Comment:
Ya, I consider since ScreenManager only visible in system app,  we probably need a wrap function like
def screen_manager_func(self):
    self.marionette.switch_to_frame()
    javascrpt_call()
    apps.switch_to_displayed_app()

So we can cover some gaia system behavior.  But I'm going to introduce big change here.